### PR TITLE
Make device event's field optional, closes #39

### LINKED ISSUE #39: Subscribe to all device events
`client.on("device-event", "my-device", fn)` should subscribe to `my-app/devices/my-device/events/#` and call `fn` on every event.

### PR DESCRIPTION
**Pull request by FokkeZB**
Also uses synon now to test callbacks

### CODE REVIEW

**Comment by FokkeZB** [comment]
Merge #50 first

**Review by johanstokking — APPROVED** [review]

**Comment by johanstokking** [comment]
@FokkeZB please rebase

File: API.md [hunk 1]
```diff
@@ -89,20 +91,37 @@ client.on('message', function cb(deviceId, data) {});
...
+      ]
+    }
+  }
+  ```
+  
+> The `activation` is a type of `device` event and internally mapped as such.
 
 ## Event: activation
 
+> **DEPRECATED:** Use the `device` event with `activations` as event filter.
```
Comment (johanstokking) [inline API.md:123]: Remove the whole thing maybe?

**Review by johanstokking — APPROVED** [review]

File: API.md (hunk 1 above)
Comment (FokkeZB) [reply API.md:123]: Since that is a breaking change it has to wait for 3.0 right?

**Pull request merged**